Robotics project configuration section extended to include templates (#2379)
o3de/o3de.org: content/docs/user-guide/interactivity/robotics/project-configuration.md

git clone https://github.com/o3de/o3de-extras
```

### Registering the Gem

To use the ROS 2 Gem in any O3DE project, you need to register it with O3DE.

For convenience, set a couple of environment variables: `O3DE_HOME` to where your O3DE is located and `O3DE_EXTRAS_HOME`
to the path of your cloned o3de-extras repository, for example:

```shell
export O3DE_HOME=${HOME}/o3de
export O3DE_EXTRAS_HOME=${HOME}/o3de-extras
```

Run the following command to register the ROS 2 Gem:
```bash
${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/ROS2
```

### Registering robotic project templates

> We also recommend that you register robotic project templates and their Asset Gems so that you can quickly start with your simulation project.
> If you would like to use templates, register them as well as their Asset Gems:

Reviewer: ```suggestion
Robot project templates can help you quickly start your simulation project. We recommend that you register the following project template Gems and their Asset Gems. 
To register robotic templates and assets:
```

> We also recommend that you register robotic project templates and their Asset Gems so that you can quickly start with your simulation project.
> If you would like to use templates, register them as well as their Asset Gems:
> 
> ```bash
> ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/ProteusRobot
> ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/RosRobotSample
> ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/WarehouseAssets
> ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/WarehouseSample
> ${O3DE_HOME}/scripts/o3de.sh register --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2FleetRobotTemplate
> ${O3DE_HOME}/scripts/o3de.sh register --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2ProjectTemplate
> ```
> 
> You will also want to make sure that you enable LFS (Large File Storage) which is used for Assets in `o3de-extras`:
> ```bash
> cd ${O3DE_EXTRAS_HOME}
> git lfs install && git lfs pull
> ```

Reviewer: - Nit: I think the part about "to quickly start your simulation project" makes a good introductory sentence, so moved that to the front.
- Listed the two commands as numerical steps.  I think enabling LFS makes sense as the first step

````suggestion
Robotics project templates can help you quickly start your simulation project. We recommend that you register the robotics project template Gems and their Asset Gems, which you downloaded with the `o3de-extras` repository.

To register robotic templates and assets:
1. Enable Git Large File Storage (LFS), if you haven't already.  Asset Gems use LFS to store large files. 
    ```bash
    cd ${O3DE_EXTRAS_HOME}
    git lfs install && git lfs pull
    ```
2. Register the following robotic templates and assets.
    ```bash
    ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/ProteusRobot
    ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/RosRobotSample
    ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/WarehouseAssets
    ${O3DE_HOME}/scripts/o3de.sh register --gem-path ${O3DE_EXTRAS_HOME}/Gems/WarehouseSample
    ${O3DE_HOME}/scripts/o3de.sh register --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2FleetRobotTemplate
    ${O3DE_HOME}/scripts/o3de.sh register --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2ProjectTemplate
    ```
````

When created with a template, a new project can start with a specific configuration. include certain enabled Gems and starting levels.

Robotic project templates are designed to help you to quickly start simulating robots in O3DE with ROS 2.

#### ROS 2 Project Templates

There are two templates for robotics:
- [ROS 2 project template](https://github.com/o3de/o3de-extras/tree/development/Templates/Ros2ProjectTemplate):
  - A versatile, lightweight template that is good for a starting project and includes a robot with differential drive.
- [Warehouse project template](https://github.com/o3de/o3de-extras/tree/development/Templates/Ros2FleetRobotTemplate):
  - A photorealistic warehouse with a Proteus robot, easy to customize and scale up (multi-robot).

:bulb: The template repositories also include examples that you can try out by following their README files.

#### Create a new project with a template

To create a project with a template, you may use GUI or command line.
The quickest way is to run the following commands (adjust `PROJECT_NAME`, `PROJECT_PATH` and the template as you wish):

```shell
export PROJECT_NAME=MySimulationProject

> export PROJECT_PATH= ${HOME}/projects/${PROJECT_NAME}

Reviewer 2: ```suggestion
export PROJECT_PATH=${HOME}/projects/${PROJECT_NAME}
```

${O3DE_HOME}/scripts/o3de.sh create-project --project-path $PROJECT_PATH --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2ProjectTemplate 
```

For more information, refer to [Project Creation](/docs/welcome-guide/create/)

### Building

The ROS 2 Gem is built when you build an O3DE project with the ROS 2 Gem enabled. 

Make sure to [source your ROS 2 workspace](#source-your-ros-2-workspace) before building.

For convenience, here is an example of parametrized CMake calls:

```shell
cd $PROJECT_PATH

> cmake -B build/linux -G "Ninja Multi-Config" -DLY_DISABLE_TEST_MODULES=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON

Reviewer 2: consider adding `-DLY_STRIP_DEBUG_SYMBOLS=ON`

Author: ```suggestion
cmake -B build/linux -G "Ninja Multi-Config" -DLY_DISABLE_TEST_MODULES=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DLY_STRIP_DEBUG_SYMBOLS=ON
```

cmake --build build/linux --config profile --target ${PROJECT_NAME} Editor ${PROJECT_NAME}.Assets 
```

For a deeper understanding on how O3DE projects are built, please refer to [Configure and Build](/docs/user-guide/build/configure-and-build).

### Launching your project in O3DE Editor

> Once your project is build, run the following command to start the Editor:

Reviewer: ```suggestion
Once your project is built, run the following command to start the Editor:
```
